tensorflow/compiler/xla/client/xla_builder.cc

```diff
@@ -2719,32 +2729,69 @@ void XlaBuilder::AddCalledComputation(const XlaComputation& computation,
   }
 }
 
-StatusOr<const HloInstructionProto*> XlaBuilder::LookUpInstruction(
-    const XlaOp& op) const {
-  TF_RETURN_IF_ERROR(first_error_);
+namespace {
 
-  if (op.builder_ == nullptr) {
+template <typename InstructionType, typename HandleToIndexType,
+          typename InstructionProtoVectorType>
+StatusOr<InstructionType> LookUpInstructionByHandleInternal(
+    HandleToIndexType& handle_to_index,
```

Comment:
Should this be `const HandleToIndexType &`

Reply:
Same for `InstructionProtoVectorType`.

Reply:
Done (requires a `const_cast` though for `InstructionProtoVectorType`)